dpcpp/CMakeLists.txt

```diff
@@ -79,7 +79,8 @@ target_link_options(ginkgo_dpcpp PRIVATE -fsycl-device-lib=all)
 
 if (NOT GINKGO_DPCPP_AOT_TARGETS STREQUAL "")
     string(REPLACE ";" "," GINKGO_DPCPP_AOT_TARGETS_FORMATTED "${GINKGO_DPCPP_AOT_TARGETS}")
-    target_compile_options(ginkgo_dpcpp PRIVATE -fsycl-targets=${GINKGO_DPCPP_AOT_TARGETS_FORMATTED})
+    target_compile_options(ginkgo_dpcpp PUBLIC -fsycl-targets=${GINKGO_DPCPP_AOT_TARGETS_FORMATTED})
+    target_link_options(ginkgo_dpcpp PUBLIC -fsycl-targets=${GINKGO_DPCPP_AOT_TARGETS_FORMATTED})
```

Comment:
Good catch on adding `link_options`. But does this need to be public? This would probably fail if some program links against Ginkgo but uses a non DPC++ compiler which does not have the option (like gcc).

Reply:
it's mainly from cpu side of https://www.intel.com/content/www/us/en/develop/documentation/oneapi-dpcpp-cpp-compiler-dev-guide-and-reference/top/compilation/ahead-of-time-compilation.html. If do not add it with the final execution, it can only use host queue (no subgroup or others support).
I am not pretty sure the compile or linking